eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxFunder.scala

```diff
@@ -426,80 +425,49 @@ private class ReplaceableTxFunder(nodeParams: NodeParams,
   }
 
   private def addInputs(anchorTx: ClaimLocalAnchorWithWitnessData, targetFeerate: FeeratePerKw, commitment: FullCommitment): Future[(ClaimLocalAnchorWithWitnessData, Satoshi)] = {
-    import fr.acinq.bitcoin.scalacompat.KotlinUtils._
+    import fr.acinq.bitcoin.scalacompat.KotlinUtils
 
     val dustLimit = commitment.localParams.dustLimit
     val commitTxWeight = commitWeight(commitment)
     // NB: fundrawtransaction requires at least one output, and may add at most one additional change output.
     // Since the purpose of this transaction is just to do a CPFP, the resulting tx should have a single change output
     // (note that bitcoind doesn't let us publish a transaction with no outputs). To work around these limitations, we
     // start with a dummy output and later merge that dummy output with the optional change output added by bitcoind.
-    val txNotFunded = anchorTx.txInfo.tx.copy(txOut = TxOut(dustLimit, Script.pay2wpkh(PlaceHolderPubKey)) :: Nil)
+    val dummyChangeOutput = TxOut(dustLimit, Script.pay2wpkh(PlaceHolderPubKey))
+    val txNotFunded = anchorTx.txInfo.tx.copy(txOut = dummyChangeOutput :: Nil)
     // The anchor transaction is paying for the weight of the commitment transaction.
-    val anchorWeight = Seq(InputWeight(anchorTx.txInfo.input.outPoint, anchorInputWeight + commitTxWeight))
-
-    def makeSingleOutputTx(fundTxResponse: OnChainWallet.FundTransactionResponse): Future[Transaction] = {
+    // We remove the weight of the artificially added change output, because we will remove that output after funding.
```

Comment:
I don't understand this change: that output will be replaced by another output of the same weight ?

Reply:
This is clearly tricky code. The weight of the anchor transaction that we want bitcoind to take into account when computing the fees is:

- the weight of the commit tx (because we want to bump it)
- the weight of the anchor input
- the weight for one change output

But we artificially add a change output here and bitcoind will add another one. After `fundrawtransaction`, we will merge those two change outputs into a single one. But that means that during `fundrawtransaction`, bitcoind thinks there are two change outputs and tries to pay the fees for the weight of those two outputs. By deducing the weight for one of them here, we ensure bitcoind is able to compute a fee that really matches the final transaction we will broadcast.

Does that make more sense?